data/data/aws/vpc/common.tf

```diff
@@ -3,7 +3,9 @@
 data "aws_region" "current" {}
 
 // Fetch a list of available AZs
-data "aws_availability_zones" "azs" {}
+data "aws_availability_zones" "azs" {
```

Comment:
Maybe I'm wrong, but with https://github.com/openshift/installer/pull/1121 the available check in go will make sure that the masterAZs chosen were available and `az_to_<public/private>subnet` maps will make sure we only create machines in AZs chosen by installer

?

Reply:
> Maybe I'm wrong, but with #1121 the available check in go...

Yup, the Terraform part of this is code that is hopefully going away soon.  But we want the Go part of this long-term, and the Terraform code is still there now, so I think we might as well fix it too.

Reply:
Well technically terraform should use the AZs given by the installer ie if the AZ goes out of service between asset generation and terraform call... it would be better if we fail with error that AZ is out of service rather than the AZ not found in the `az_to_<public/private>subnet` maps.

Reply:
> Well technically terraform should use the AZs given by the installer...

Agreed.

> ... ie if the AZ goes out of service between asset generation and terraform call...

I think using Go-provided zones means we can drop this `data "aws_availability_zones"` section altogether.  Guarding against a zone going down is always going to be racy up through actually creating resources in the zone, so I'd rather save a few lines and the mental overhead and just have Terraform assume Go has already done any required validation and blindly consume what it was given.

Reply:
> I think using Go-provided zones means we can drop this `data "aws_availability_zones"` section altogether. Guarding against a zone going down is always going to be racy up through actually creating resources in the zone, so I'd rather save a few lines and the mental overhead and just have Terraform assume Go has already done any required validation and blindly consume what it was given.

I can get behind that change. :)

Reply:
> I can get behind that change. :)

Did you want me to pick that up in this PR?  Or submit it as a follow-up once this lands?

Reply:
> > I think using Go-provided zones means we can drop this `data "aws_availability_zones"` section altogether. Guarding against a zone going down is always going to be racy up through actually creating resources in the zone, so I'd rather save a few lines and the mental overhead and just have Terraform assume Go has already done any required validation and blindly consume what it was given.
> 
> I can get behind that change. :)

But that would need some changes esp https://jira.coreos.com/browse/CORS-902, that will push this PR to 4.1

Reply:
> > I can get behind that change. :)
> 
> Did you want me to pick that up in this PR? Or submit it as a follow-up once this lands?

follow up is fine.